Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var center = [this._object.x + this._radius, this._object.y + this._radius];
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.hitBoxes;
+  });
+  hitBoxes.push(this._object.getHitBoxes());
```

Comment:
Why adding the hit boxes of the object? The object *is* the light, so it should not be taken into account of the *light* polygon because... well because it's the light itself :D 
The rays are emitted from the point (`this._object.x`, `this._object.x`) are are stopped by the obstacles around (`this._manager.getAllObstaclesAround`). You only care about the hitboxes of these objects.

Reply:
> Why adding the hit boxes of the object?

We'll need to add the hitbox of the object because we need to know the intersection points on the boundary of mesh as well, to generate a complete geometry.
![Screenshot (2)](https://user-images.githubusercontent.com/39851078/84774953-2a07e000-affc-11ea-9e5d-3414b415b9e1.png)

In this article https://ncase.me/sight-and-light/ (and the other articles as well) you can see that points of intersection with the boundary of light is required to create triangles.

Reply:
As discussed that's indeed required in our algorithm, good spot!